### DISCUSSION TITLE
Exploration du schéma Métopes

### DISCUSSION THREAD

**Discussion by emchateau** [ouvroir/encyclopedie#34] [General]
Métopes vient de rendre public un schéma réalisé dans le cadre du projet FNSO.
- https://git.unicaen.fr/fnso/i-fair-ir/tei-commons-publishing
- doc https://tei-commons-publishing.readthedocs.io/fr/latest/odd_doc.html

Après une rapide évaluation préliminaire, il semble que le schéma soit suffisamment générique pour être utilisé dans le cadre de notre projet. Il y a peu d’ajustements à faire sur les articles cibles pour valider.

Nota, le modèle utilise un système de chaînage pour décliner des sous-modèles.

**Comment by emchateau** [comment]
D’après une première évaluation, les éléments suivants devraient peut-être ajustés pour nos besoins
- ajouter pour rend les valeurs : `smallcaps` (il y a seulement small-caps), `superscript`, `underscript`
- ajouter l’élément `<q>`
- ajouter les entités-nommées `<orgName>`, `<persName>`, `<objectName>`, `<placeName>`

**Reply by pippaJeanne** [reply]
Est-ce qu'il ya vraiment besoin d'ajouter ces valeurs pour rend ? C'est question d'utiliser leur version : small-caps = smallcaps, sup = superscript, sub = underscript.

orgName, persName et placeName sont pris en compte. Il ne faudrait qu'ajouter objectName.

**Reply by pippaJeanne** [reply]
placeName n'était pas pris en compte.

**Comment by emchateau** [comment]
D’après une première exploration du modèle, on valide correctement l’entête TEI avec `tei-commons-publishing/rng/commons-publishing-openedition.rng` mais pas avec `tei-commons-publishing/rng/commons-publishing-metopes.rng`

Est-ce que le schéma métopes est encore incomplet ? Ne devrions-nous pas chaîner un nouveau modèle depuis openedition pour le spécialiser avec nos besoins ?

**Reply by emchateau** [reply]
cc. @pippaJeanne

**Reply by pippaJeanne** [reply]
Effectivement, le schéma openedition semble être plus complet.

**Comment by emchateau** [comment]
À mon avis, ils font erreur dans leur manière de chaîner les schémas. 
Ils utilisent le mécanisme comme s’il s’agissait de déterminer un héritage à partir du document maître. 
Or, ce n’est pas à ma connaissance la logique normalement mise en œuvre par la TEI. Il y a normalement un schéma source qui déclare tous les éléments valides dans une classe de documents. Mais chaque sous-modèles doit également importer toutes les définitions dont il a besoin.
Utilisons le modèle complet pour le moment. Je pense qu’il faudrait leur poser la question.

**Comment by pippaJeanne** [comment]
@emchateau  J'ai fait une version de schéma en utilisant le chaînage à partir du schéma openedition en ajoutant les specifications dont on avait parlé. Le système de chaînage peut vraiment faire l'affaire je crois.

**Reply by emchateau** [reply]
Merci, 
As-tu poussé ton travail quelque part que je puisse regarder le balisage, stp ?

**Reply by pippaJeanne** [reply]
C'est le fichier [nucma_custom.odd](https://github.com/ouvroir/encyclopedie/blob/main/prototype/odd/nucma_custom.odd).
Et tu peux aussi utiliser le schéma [nucma_custom.rng](https://github.com/ouvroir/encyclopedie/blob/main/prototype/schemas/nucma_custom.rng).

J'ai pas modifié les valeurs de rend par contre. C'est pris en compte par le schéma d'openedition, il faudrait juste s'adapter à leur terminologie comme j'avait expliqué précédemment.
